go/adbc/driver/databricks/command.go

```diff
@@ -0,0 +1,384 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package databricks
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/apache/arrow-adbc/go/adbc"
+	"github.com/apache/arrow-go/v18/arrow"
+	"github.com/apache/arrow-go/v18/arrow/array"
+	"github.com/apache/arrow-go/v18/arrow/memory"
+	"github.com/databricks/databricks-sdk-go/service/compute"
+)
+
+const (
+	CmdOptionIntByteLimit  = "adbc.databricks.byte_limit"
+	CmdOptionIntRowLimit   = "adbc.databricks.row_limit"
+	CmdOptionBoolTruncated = "adbc.databricks.truncated" // read-only
+)
+
+type command struct {
+	alloc memory.Allocator
+	conn  *connectionImpl
+
+	req *compute.Command
+
+	commandId string
+}
+
+func NewCommand(conn *connectionImpl) (adbc.Statement, error) {
+	cluster, err := conn.client.Clusters.Get(context.Background(), compute.GetClusterRequest{
+		ClusterId: conn.client.Config.ClusterID,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if cluster.State == compute.StateTerminated || cluster.State == compute.StateTerminating {
+		wait, err := conn.client.Clusters.Start(context.Background(), compute.StartCluster{
+			ClusterId: conn.client.Config.ClusterID,
+		})
+		if err != nil {
+			return nil, err
+		}
+		_, err = wait.Get()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &command{
+		alloc: conn.Alloc,
+		conn:  conn,
+		req: &compute.Command{
+			Command:   "",
+			Language:  compute.LanguageSql,
+			ClusterId: conn.client.Config.ClusterID,
+			ContextId: fmt.Sprint(time.Now().UnixNano()),
```

Comment:
What is this for? Should we have one per connection, instead?

Reply:
this is actually a really good catch, this is the ID of an "execution context" (essentially a single REPL environment for your execution language). We should be asking databricks to create this and _I think_ reusing this for a connection

Reply:
REST docs: https://docs.databricks.com/api/workspace/commandexecution/create

Reply:
If you generate an UUID at connection time and re-use it, it will be unique. No need to ask that Databricks create it. The apocalypse is more probably than a collision on a random 128-bit identifier.

Reply:
seems like the context Id needs to be a numeric string and in this case it will be more performant for a connection to re-use the existing context we have asked them to create instead of creating it on the fly

Reply:
seems like it will get created on first statement execution if you don't ask it to be created ahead of time